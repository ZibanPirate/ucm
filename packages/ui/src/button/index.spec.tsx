import { fireEvent, render, screen } from "@testing-library/react";

import { Button } from ".";

describe(`Testing component '${Button.name}' :`, () => {
  it(`should render with text 'OK' and fire an 'onClick' event when clicked`, () => {
    const onClick = jest.fn();
    const { container } = render(
      <Button data-testid="ok-button" onClick={onClick}>
        OK
      </Button>,
    );
    expect(container).toMatchSnapshot();

    const okButton = screen.getByTestId("ok-button");
    fireEvent.click(okButton);
    expect(onClick).toBeCalledTimes(1);
  });

  it(`should render a stretched '${Button.name}' component with text 'See More' and do nothing when clicked`, async () => {
    const { container } = render(
      <Button data-testid="see-more-button" stretch>
        See More
      </Button>,
    );
    expect(container).toMatchSnapshot();

    const seeMoreButton = screen.getByTestId("see-more-button");

    fireEvent.click(seeMoreButton);
  });
});
