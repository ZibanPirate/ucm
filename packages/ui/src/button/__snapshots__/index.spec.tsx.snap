// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component 'Button' : should render a stretched 'Button' component with text 'See More' and do nothing when clicked 1`] = `
<div>
  <button
    class="ucm-ui-button"
    data-testid="see-more-button"
    style="font-size: medium; flex: 1; margin: initial;"
  >
    See More
  </button>
</div>
`;

exports[`Testing component 'Button' : should render with text 'OK' and fire an 'onClick' event when clicked 1`] = `
<div>
  <button
    class="ucm-ui-button"
    data-testid="ok-button"
    style="font-size: medium; margin: initial;"
  >
    OK
  </button>
</div>
`;
